### PR TITLE
The Sleeping Carp Vampires

### PR DESCRIPTION
**Pull request by Twinmold93**
There is a small chance someone who is a vampire will also get traitor, or vice versa. If they get the martial art scroll (sleeping carp), they are unable to harm intent the head to drink blood anymore. This makes it so they can if they have sleeping carp.

:cl: Twinmold
Fix: Vampires with sleeping carp (traitor+vampire rounds) can now suck people's blood like normal.
/:cl:

### CODE REVIEW

File: code/modules/martial_arts/sleeping_carp.dm [hunk 1]
```diff
@@ -113,6 +113,8 @@
 
 /datum/martial_art/the_sleeping_carp/harm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
 	add_to_streak("H",D)
+	if((A.mind in ticker.mode.vampires)&(A.zone_sel.selecting == "head"))
```
Comment (Fox-McCloud) [inline code/modules/martial_arts/sleeping_carp.dm]: spacing and shouldn't this be &&

**Comment by Fox-McCloud** [comment]
Remind me again why blood-sucking is snowflaked into `attack_hand` instead of being a spell.

**Comment by Twinmold93** [comment]
I'd say ask the person who made vampires. I suppose to try and make it a little bit faster for vampires to start draining blood.

**Comment by FreeStylaLT** [comment]
That must have


sucked

**Comment by Regen1** [comment]
>Remind me again why blood-sucking is snowflaked into attack_hand instead of being a spell.

"So if you're truly not a vampire, punch me in the head!"

**Comment by Krausus** [comment]
Nope. Don't snowflake this into Sleeping Carp. Go to [where martial arts and vampire biting is handled](https://github.com/ParadiseSS13/Paradise/blob/71e5344a983c85cd9bc315868ec9f869770f28b7/code/modules/mob/living/carbon/human/human_attackhand.dm#L111-L140) and make vampire biting take priority there.

**Comment by Lady-Luck** [comment]
And another thing, why can't vampires bite someone wearing gas mask?

I can't think of any mask that would render the neck completely inaccessible.

**Comment by Twinmold93** [comment]
Moved the priority in attack to vampire bite first, then the check for attack. Removed the additional code in the sleeping_carp.dm. Makes it so it'll work with all martial arts this way, as suggested by @Krausus

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 2]
```diff
@@ -109,36 +109,34 @@
...
+					to_chat(M, "<span class='warning'>There is no blood in a skeleton!</span>")
+					return 0
+				if(issmall(src) && !ckey) //Monkeyized humans are okay, humanized monkeys are okey, monkeys are not.
+					to_chat(M, "<span class='warning'>Blood from a monkey is useless!</span>")
+					return 0
+				//we're good to suck the blood, blaah
+				M.mind.vampire.handle_bloodsucking(src)
+				add_logs(src, M, "vampirebit")
+				msg_admin_attack("[key_name_admin(M)] vampirebit [key_name_admin(src)]")
+				return 0
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/human_attackhand.dm]: I believe this should be a return `return`

**Comment by Twinmold93** [comment]
Seems that none of those zeros were actually necessary in this modification.

**Pull request merged**